app/presenters/subscription_presenter.rb

```diff
@@ -1,11 +1,30 @@
 class SubscriptionPresenter < BasePresenter
   include ApplicationHelper
 
+  SEARCH_CRITERIA_SORT_ORDER = %w[location radius keyword subject job_title minimum_salary maximum_salary
+                                  working_pattern phases newly_qualified_teacher].freeze
+
   def filtered_search_criteria
-    @filtered_search_criteria ||= search_criteria_to_h.each_with_object({}) do |(field, value), criteria|
-      search_field = add_search_criteria_field(field, value)
-      criteria.merge!(search_field) if search_field.present?
-    end.stringify_keys
+    @filtered_search_criteria ||= begin
+                                    criteria = {}
+                                    sorted_criteria = search_criteria_to_h.sort do |(a, _), (b, _)|
```

Comment:
I reckon it'd be nicer to break this out into a method to be clearer about the intent here. Would also make it nicely testable to catch any edge cases

Reply:
:+1: